analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantModifierCodeFix.cs

```diff
@@ -62,117 +55,52 @@ protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext c
             }
-
-            if (node is CheckedExpressionSyntax checkedExpression)
+            else if (token.Parent is CheckedExpressionSyntax checkedExpression)
             {
-                context.RegisterCodeFix(
-                    CodeAction.Create(
-                        TitleChecked,
-                        c =>
-                        {
-                            var newRoot = RemoveRedundantCheckedExpression(root, checkedExpression);
-                            return Task.FromResult(context.Document.WithSyntaxRoot(newRoot));
-                        }),
-                    context.Diagnostics);
+                context.RegisterCodeFix(CodeAction.Create(TitleChecked, c => ReplaceRoot(context, RemoveRedundantCheckedExpression(root, checkedExpression))), context.Diagnostics);
             }
+            return Task.CompletedTask;
         }
 
+        private static Task<Document> ReplaceRoot(CodeFixContext context, SyntaxNode newRoot) =>
+            Task.FromResult(context.Document.WithSyntaxRoot(newRoot));
+
         private static SyntaxNode RemoveRedundantUnsafe(SyntaxNode root, SyntaxToken token)
         {
-            if (!(token.Parent is UnsafeStatementSyntax unsafeStatement))
+            if (token.Parent is UnsafeStatementSyntax unsafeStatement)
             {
-                return RemoveRedundantToken(root, token);
+                return unsafeStatement.Parent is BlockSyntax parentBlock && parentBlock.Statements.Count == 1
+                    ? root.ReplaceNode(parentBlock, parentBlock.WithStatements(unsafeStatement.Block.Statements).WithAdditionalAnnotations(Formatter.Annotation))
+                    : root.ReplaceNode(unsafeStatement, unsafeStatement.Block.WithAdditionalAnnotations(Formatter.Annotation));
             }
-
-            if (unsafeStatement.Parent is BlockSyntax parentBlock &&
-                parentBlock.Statements.Count == 1)
+            else
             {
-                return root.ReplaceNode(
-                    parentBlock,
-                    parentBlock.WithStatements(unsafeStatement.Block.Statements).WithAdditionalAnnotations(Formatter.Annotation));
+                return RemoveRedundantToken(root, token);
             }
-
-            return root.ReplaceNode(
-                unsafeStatement,
-                unsafeStatement.Block.WithAdditionalAnnotations(Formatter.Annotation));
         }
 
         private static SyntaxNode RemoveRedundantToken(SyntaxNode root, SyntaxToken token)
         {
             var oldParent = token.Parent;
-            var newParent = oldParent.ReplaceToken(
-                token,
-                SyntaxFactory.Token(SyntaxKind.None));
-
-            return root.ReplaceNode(
-                oldParent,
-                newParent.WithLeadingTrivia(oldParent.GetLeadingTrivia()));
+            var newParent = oldParent.ReplaceToken(token, SyntaxFactory.Token(SyntaxKind.None));
```

Comment:
NIT: After placing a `using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;` at the top of the file all the `SyntaxFactory.` snippets can be removed.

Reply:
It's just two cases, I'd leave it as it is here.